vmdk_plugin/refcnt.go

```diff
@@ -150,18 +169,24 @@ func (r refCountsMap) Init(d *vmdkDriver) {
 		return
 	}
 
-	log.Infof("Discovered %d volumes in use.", len(r))
-	for name, cnt := range r {
+	// RLocks the refCountsMap
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+	log.Infof("Discovered %d volumes in use.", len(r.refMap))
+	for name, cnt := range r.refMap {
 		log.Infof("Volume name=%s count=%d mounted=%t device='%s'",
 			name, cnt.count, cnt.mounted, cnt.dev)
 	}
 }
 
 // Returns ref count for the volume.
 // If volume is not referred (not in the map), return 0
-// NOTE: this assumes the caller holds the lock if we run concurrently
 func (r refCountsMap) getCount(vol string) uint {
-	rc := r[vol]
+	// RLocks the refCountsMap
+	r.mtx.RLock()
```

Comment:
The docker daemon has a reference counted resource locking by name scheme (much like what we have in the esx service) that serialize operations per volume (pkg/locker and volume/store) so we can't have more than one op on the same daemon/volume at the same time, so race on the refcount value is not possible. 
Our only concern it's to serialize reads/writes on the refcount map.